### DISCUSSION TITLE
Error resolving dependencies when upgrading to v10

### DISCUSSION THREAD

**Discussion by wmonecke-gen** [maplibre/maplibre-react-native#546] [Q&A] [answered]
While trying to upgrade to v10 I am encountering issues in xCode.

![Image](https://github.com/user-attachments/assets/ce29aa98-4a62-4c56-96ad-f04634cff1af)

```bash
Failed to resolve dependencies Dependencies could not be resolved because root depends on 'maplibre-gl-native-distribution' 6.5.4..<7.0.0 and root depends on 'maplibre-gl-native-distribution' 5.13.0.
```

I followed the installation steps here:
https://github.com/maplibre/maplibre-react-native/blob/main/ios/install.md#ios-installation

Changed the `post_install` to `$RCTMLN.post_install(installer)`

and even added the following to the Podfile:

```bash
$RCTMLN_SPM_Spec = {
  url: "https://github.com/maplibre/maplibre-gl-native-distribution",
  requirement: {
    kind: "upToNextMajorVersion",
    minimumVersion: "6.5.4"
  },
  product_name: "MapLibre"
}
```

Any ideas what could be going on?

Seems like something from the previous version is still hanging around somehow?

### Environment

- Platform: iOS
- Emulator or Simulator: no
- Development OS: macOS 
- `@maplibre/maplibre-react-native` Version: "10.0.0-alpha.26"
- `react-native` Version: "0.74.2"

**Comment by wmonecke-gen** [comment, marked as answer]
I fixed it by following these steps and setting the exact version to the one in the error.

https://i.sstatic.net/nZ4vf.jpg

Still a bit weird it was there after removing and reinstalling.

**Comment by wmonecke-gen** [comment]
I am now getting the following error though:

![Image](https://github.com/user-attachments/assets/00468127-9a06-464d-a3f1-bc99772e006c)

This is my `Package.resolved`

```bash
{
  "originHash" : "93004ae322b4ca0d0df41da4297c41af20f08ba47df8cd8af368bad1187ee8c2",
  "pins" : [
    {
      "identity" : "geos",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/GEOSwift/geos.git",
      "state" : {
        "revision" : "f510e634c822116fca615064d889300dba40d761",
        "version" : "8.1.0"
      }
    },
    {
      "identity" : "geoswift",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/GEOSwift/GEOSwift",
      "state" : {
        "revision" : "939c9e7f19a084a425239dd8e2c744beb78da7d6",
        "version" : "10.3.0"
      }
    },
    {
      "identity" : "maplibre-gl-native-distribution",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/maplibre/maplibre-gl-native-distribution",
      "state" : {
        "revision" : "abe762f1e19e03a4c6943d2aad92c219da384b29",
        "version" : "6.5.4"
      }
    }
  ],
  "version" : 3
}
```

And yeah Mapbox is not being included, but its still being requested somehow

In xCode `Link Binaries with Libraries` I do see Mapbox, should that even be there?

![Image](https://github.com/user-attachments/assets/f0c8e98d-1324-47f0-865e-b8f4bb813b30)

If I remove Mapbox from the Linked Binaries I get the following error:

![Image](https://github.com/user-attachments/assets/98ae7a37-6824-440c-816b-395a509dcb2b)

**Comment by wmonecke-gen** [comment]
@KiwiKilian do you have any ideas on this one?

**Comment by KiwiKilian** [comment]
Looks quite a few individual problems with your setup. Mapbox probably shouldn't be required anymore.

**Comment by wmonecke-gen** [comment]
@KiwiKilian I finally managed to solve the other errors.

However, I still see an error when using `.addCustomHeaders` in v10 alpha.27. This is the error:

 ![Image](https://github.com/user-attachments/assets/f832e77d-75dc-4066-b82b-fe20c2be7b99)

This is our `AppDelegate.m`

```swift
#import "AppDelegate.h"
#import "RNBootSplash.h" // https://github.com/zoontek/react-native-bootsplash

// react-native-firebase, https://rnfirebase.io/#configure-firebase-with-ios-credentials
#import <Firebase.h>
// https://github.com/maplibre/maplibre-react-native/blob/main/docs/CustomHttpHeaders.md#custom-http-headers
#import "MLNCustomHeaders.h"
// react-native-background-fetch. https://github.com/transistorsoft/react-native-background-fetch/blob/master/docs/INSTALL-AUTO-IOS.md#appdelegate
#import <TSBackgroundFetch/TSBackgroundFetch.h>

#import <React/RCTBridge.h>
#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>


@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  self.moduleName = @"AppName";
  // You can add your custom initial props in the dictionary below.
  // They will be passed down to the ViewController used by React Native.
  self.initialProps = @{};

  [FIRApp configure];
  
  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
                                                   moduleName:@"AppName"
                                            initialProperties:nil];
  
  if (@available(iOS 13.0, *)) {
    rootView.backgroundColor = [UIColor systemBackgroundColor];
  } else {
    rootView.backgroundColor = [UIColor whiteColor];
  }
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];

  // maplibre-react-native
  [[MLNCustomHeaders sharedInstance] initHeaders];

  // react-native-background-fetch
  [[TSBackgroundFetch sharedInstance] didFinishLaunching];

  return [super application:application didFinishLaunchingWithOptions:launchOptions];
}
```

Is this an issue with our configuration?

**Reply by KiwiKilian** [reply]
This actually was a bug in the lib #551, now fixed in `10.0.0-beta.8`

**Comment by KiwiKilian** [comment]
Please provide a minimal reproduction of your problem. Otherwise I still can only suspect your project setup.
